src/DataProvider/UserCollectionDataProvider.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\DataProvider;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Paginator;
+use ApiPlatform\Core\DataProvider\ContextAwareCollectionDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use App\Entity\User;
+use App\Repository\UserRepository;
+use App\Utils\ValidationUtils;
+use Doctrine\ORM\Tools\Pagination\Paginator as DoctrinePaginator;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Security;
+
+final class UserCollectionDataProvider implements ContextAwareCollectionDataProviderInterface, RestrictedDataProviderInterface
+{
+    public function __construct(
+        private readonly RequestStack $requestStack,
+        private readonly UserRepository $userRepository,
+        private readonly ValidationUtils $validationUtils,
+        private readonly Security $security,
+    ) {}
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return User::class === $resourceClass && 'get' === $operationName;
+    }
+
+    public function getCollection(string $resourceClass, string $operationName = null, array $context = []): Paginator
+    {
+        $currentRequest = $this->requestStack->getCurrentRequest();
+        $itemsPerPage = $currentRequest->query?->get('itemsPerPage') ?? 10;
+        $page = $currentRequest->query?->get('page') ?? 1;
+
+        /** @var User $currentUser */
+        $currentUser = $this->security->getUser();
+        $activeTenant = $currentUser->getActiveTenant();
+        $activeTenantUlid = $this->validationUtils->validateUlid($activeTenant->getId());
+
+        $queryBuilder = $this->userRepository->getExternalUsersByTenantQueryBuilder($activeTenantUlid);
```

Comment:
It's misleading that a `UserCollectionDataProvider` only returns external users. At a minimum rename the class.

Reply:
Not sure how to best handle this on the API level. on the one hand I would prefer one endpoint that can be filtered. On the other hand I'm not sure how clean an implementation that is in regards to access control. We would then on the security level have to enforce that the endpoint is called with the correct filter.

Reply:
I have removed the external part.
I have made a doctrine extension to limit access.